Add support for Mod Settings

utoxin opened the pull request:
Adds support for the Mod Settings screen

utoxin committed:
Add support for Mod Settings

Adds support for the Mod Settings screen

utoxin commented:
This makes PR #10 obsolete, and resolves both issues #8 and #9.

jaguilar commented on long-reach_0.0.7/control.lua:
```diff
@@ -0,0 +1,22 @@
+function sync_long_reach_settings()
+		local settings = settings.global
+		
+		-- Default build and reach distances are 6, so subtract that from the requested distance
```
This kind of comment can be elided by using a local variable.

local default_action_distance = 6
... - default_action_distance

jaguilar commented on long-reach_0.0.7/control.lua:
```diff
@@ -0,0 +1,22 @@
+		-- Default build and reach distances are 6, so subtract that from the requested distance
+		game.forces["player"].character_build_distance_bonus = settings["long-reach-build-distance"]["value"] - 6
+		game.forces["player"].character_reach_distance_bonus = settings["long-reach-reach-distance"]["value"] - 6
+end
```
Conventionally, we have newlines at the end of files. It might be best to change your editor to add these automatically. Please add one here and elsewhere.

jaguilar reviewed (changes requested):
Just a couple of minor nitpicks. I have to test this myself tomorrow, but everything appears to be in order.

utoxin committed:
Style Cleanup

Cleaning up code style per comments.
Also fixing tabbing on method. Bad editor. Bad.

utoxin commented:
Sorry about that! Also fixed the bad indentation on the last method. Not using a full blown IDE, since I didn't have a Lua IDE handy, and Notepad++ doesn't handle things very cleanly.

utoxin committed:
Missed a line ending

Missed saving this line ending change.

utoxin committed:
Another missed save

I really hate notepad++
